src/main/java/org/broadinstitute/hellbender/utils/clipping/ReadClipper.java

```diff
@@ -488,10 +492,146 @@ protected GATKRead hardClipByReferenceCoordinates(final int refStart, final int
         if ( start > 0 && stop < read.getLength() - 1) {
             throw new GATKException(String.format("Trying to clip the middle of the read: start %d, stop %d, cigar: %s", start, stop, read.getCigar().toString()));
         }
-
         this.addOp(new ClippingOp(start, stop));
         final GATKRead clippedRead = clipRead(ClippingRepresentation.HARDCLIP_BASES);
         this.ops = null;
         return clippedRead;
     }
+
+
+
+    /**
+     * Soft clip the read to the variable region (from refStart to refStop) processing also the clipped bases
+     *
+     * @param read     the read to be clipped
+     * @param refStart the beginning of the variant region (inclusive)
+     * @param refStop  the end of the variant region (inclusive)
+     * @return the read soft clipped to the variant region
+     */
+    public static GATKRead softClipToRegionIncludingClippedBases( final GATKRead read, final int refStart, final int refStop ) {
+        final int start = read.getUnclippedStart();
+        final int stop = start + CigarUtils.countRefBasesBasedOnCigar(read, 0, read.numCigarElements()) - 1;
+
+        if (start <= refStop && stop >= refStart) {
+            if (start < refStart && stop > refStop) {
+                return (new ReadClipper(read)).softClipBothEndsByReferenceCoordinates(refStart - 1, refStop + 1);
+            } else if (start < refStart) {
+                return (new ReadClipper(read)).softClipByReferenceCoordinates(-1, refStart - 1);
+            } else if (stop > refStop) {
+                return (new ReadClipper(read)).softClipByReferenceCoordinates(refStop + 1, -1);
+            }
+            return read;
+        } else {
+            return ReadUtils.emptyRead(read);
+        }
+    }
+
+
+    /**
+     * Soft clips both tails of a read.
+     *   Left tail goes from the beginning to the 'left' coordinate (inclusive)
+     *   Right tail goes from the 'right' coordinate (inclusive) until the end of the read
+     *
+     * @param left the coordinate of the last base to be clipped in the left tail (inclusive)
+     * @param right the coordinate of the first base to be clipped in the right tail (inclusive)
+     * @return a new read, without the clipped bases
+     */
+    private GATKRead softClipBothEndsByReferenceCoordinates(final int left, final int right) {
+        if (read.isEmpty() || left == right) {
+            return ReadUtils.emptyRead(read);
+        }
+        final GATKRead leftTailRead = softClipByReferenceCoordinates(right, -1);
+
+        // after clipping one tail, it is possible that the consequent hard clipping of adjacent deletions
```

Comment:
soft -> hard? or are you saying that adjacent deletions are hard clipped regardless of whether you're soft clipping or hard clipping?